workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestHelper.cs

```diff
@@ -0,0 +1,220 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct Interest component snapshots.
+    /// </summary>
+    public class InterestHelper
+    {
+        private readonly Dictionary<uint, ComponentInterest> interest;
+
+        private InterestHelper(Dictionary<uint, ComponentInterest> interest)
+        {
+            this.interest = interest ?? new Dictionary<uint, ComponentInterest>();
+        }
+
+        /// <summary>
+        ///     Creates a new InterestHelper object.
+        /// </summary>
+        /// <returns>
+        ///     A new InterestHelper object.
+        /// </returns>
+        public static InterestHelper Begin()
+        {
+            return new InterestHelper(null);
+        }
+
+        /// <summary>
+        ///     Creates a new InterestHelper object given an existing Interest component.
+        /// </summary>
+        /// <param name="interest">
+        ///     An existing Interest component.
+        /// </param>
+        /// <returns>
+        ///     An InterestHelper object.
+        /// </returns>
+        public static InterestHelper From(Interest.Component interest)
+        {
+            return new InterestHelper(interest.ComponentInterest
```

Comment:
Dictionary has a copy constructor, use that instead

Reply:
This does a ShallowCopy
Make sure it copies over the internal List of `Query` as well, and that the Query copy makes a deepcopy of its internals as well.